tests/test_xtarfile.py

```diff
@@ -1,103 +1,205 @@
 from io import BytesIO
-from itertools import product
```

Comment:
Would it be possible to first change the implementation but keep the previous tests so that we can ensure backwards compatibility of the refactor? I'd be happy to accept a pull request for the tests after the refactor is merged.

Reply:
I tried running the original test and it works fine with a few extra lines of code in `xtarfile.py` that don't change anything about how it works. So I'll split off the tests from this PR.